src/controllers/blocks/BlocksExtrinsicsController.ts

```diff
@@ -62,10 +66,23 @@ export default class BlocksExtrinsicsController extends AbstractController<Block
 			paraId: undefined,
 		};
 
+		const cacheKey =
+			hash.toString() +
+			Number(eventDocs) +
```

Comment:
For the use of `Number' here I think it would be incorrect to pass in the args as strings - but instead would need to be booleans. Example: `eventDocs` should be `eventDocsArg`

Reply:
I agree with Tarik here. We should not pass a string since it will always result to `NaN`. You can also use `options.eventDocs`.

Reply:
updated